src/Microsoft.Azure.ServiceBus/Amqp/ActiveClientLinkManager.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.Amqp
+{
+    using Microsoft.Azure.Amqp;
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    sealed class ActiveClientLinkManager
+    {
+        static readonly TimeSpan SendTokenTimeout = TimeSpan.FromMinutes(1);
+        static readonly TimeSpan TokenRefreshBuffer = TimeSpan.FromSeconds(10);
+
+        readonly string clientId;
+        readonly ICbsTokenProvider cbsTokenProvider;
+        readonly Timer sendReceiveLinkCBSTokenRenewalTimer;
+        readonly Timer requestResponseLinkCBSTokenRenewalTimer;
+
+        ActiveSendReceiveClientLink activeSendReceiveClientLink;
+        ActiveRequestResponseLink activeRequestResponseClientLink;
+
+        public ActiveClientLinkManager(string clientId, ICbsTokenProvider tokenProvider)
+        {
+            this.clientId = clientId;
+            this.cbsTokenProvider = tokenProvider;
+            this.sendReceiveLinkCBSTokenRenewalTimer = new Timer(OnRenewSendReceiveCBSToken, this, Timeout.Infinite, Timeout.Infinite);
+            this.requestResponseLinkCBSTokenRenewalTimer = new Timer(OnRenewRequestResponseCBSToken, this, Timeout.Infinite, Timeout.Infinite);
+        }
+
+        public void Close()
+        {
+            this.ChangeRenewTimer(this.activeSendReceiveClientLink, Timeout.InfiniteTimeSpan);
+            this.ChangeRenewTimer(this.activeRequestResponseClientLink, Timeout.InfiniteTimeSpan);
+        }
+
+        public void SetActiveSendReceiveLink(ActiveSendReceiveClientLink sendReceiveClientLink)
+        {
+            this.activeSendReceiveClientLink = sendReceiveClientLink;
+            this.activeSendReceiveClientLink.Link.Closed += new EventHandler(this.OnSendReceiveLinkClosed);
+            if (this.activeSendReceiveClientLink.Link.State == AmqpObjectState.Opened)
+            {
+                this.SetRenewCBSTokenTimer(sendReceiveClientLink);
+            }
+        }
+
+        public void SetActiveRequestResponseLink(ActiveRequestResponseLink requestResponseLink)
+        {
+            this.activeRequestResponseClientLink = requestResponseLink;
+            this.activeRequestResponseClientLink.Link.Closed += new EventHandler(this.OnRequestResponseLinkClosed);
+            if (this.activeRequestResponseClientLink.Link.State == AmqpObjectState.Opened)
+            {
+                this.SetRenewCBSTokenTimer(requestResponseLink);
+            }
+        }
+
+        void OnSendReceiveLinkClosed(object sender, EventArgs e)
+        {
+            this.ChangeRenewTimer(this.activeSendReceiveClientLink, Timeout.InfiniteTimeSpan);
```

Comment:
Ack. I dont need to do this when Links are closed, because then the timers will need to be recreated if the link is recreated. But I should definitely do this on Close().. Will address in the next PR..